### ISSUE TITLE
Associated types of TryFuture and TryStream

### ISSUE THREAD

**Issue by MajorBreakfast** [rust-lang-nursery/futures-rs#1090]
Currently we have:
```rust
pub trait Future { type Output; ... }
pub trait TryFuture { type Item; type Error; ... }
```

```rust
pub trait Stream { type Item; ... }
pub trait TryStream { type TryItem; type TryError; ... }
```

- `TryStream`'s success associated type cannot be called `Item` because `Stream` already uses that name for its associated type. That's why we end up with the not so great names `TryItem` and `TryError`
- `TryFuture` and `TryStream` have inconsistently named associated types

Here's what I propose:
```rust
pub trait Future { type Output; ... }
pub trait TryFuture { type Ok; type Err; ... }
```

```rust
pub trait Stream { type Item; ... }
pub trait TryStream { type Ok; type Err; ... }
```

That way the naming would be consistent with the variants of the `Result` enum!

```rust
Future<Output = Result<T, E>>
TryFuture<Ok = T, Err = E>
```

```rust
Stream<Item = Result<T, E>>
TryStream<Ok = T, Err = E>
```

**Comment by MajorBreakfast** [comment]
Maybe `Error` instead of `Err` to align with the [`Try` trait](https://doc.rust-lang.org/std/ops/trait.Try.html)

**Comment by MajorBreakfast** [comment]
Implemented by #1094
